### PR TITLE
Fixing asgrard

### PR DESCRIPTION
**Pull request by awais786**
Fixing asgard tests.

[Ticket information](https://github.com/openedx/tubular/issues/629)]

**PART 2**

### CODE REVIEW

File: tubular/admin/boto_to_add_ingress.py [hunk 1]
```diff
@@ -45,90 +45,104 @@ def add_ingress_rule(dry_run, go_agent_security_group, go_agent_security_group_o
         - If you don't know what this is, ask someone in DevOps.
     - THEN run this script.
     """
-    asg_conn = AutoScaleConnection()
-    ec2_conn = boto.ec2.connect_to_region('us-east-1')
+    asg_conn = boto3.client('autoscaling')
```
Comment (awais786) [inline tubular/admin/boto_to_add_ingress.py:48]: This file has lots changes which need review from domain expert.

File: tubular/tests/test_boto_to_add_ingress.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+"""
```
Comment (awais786) [inline tubular/tests/test_boto_to_add_ingress.py:1]: I have added this test just to execute the [tubular/admin/boto_to_add_ingress.py](https://github.com/openedx/tubular/pull/655/files#diff-0582132c19a4ca8f0c4095692cc44e5bdc3d9637632a827e1cf047702e86283b)

File: tubular/admin/boto_to_add_ingress.py [hunk 3]
```diff
@@ -45,90 +45,105 @@ def add_ingress_rule(dry_run, go_agent_security_group, go_agent_security_group_o
...
             ))
             raise
 
     # Construct a fake security group for the prod-tools-goagent-sg security group in the edx-tools account.
     # This group will be used to grant the go-agents ingress into the ASG's VPCs.
-    go_agent_security_group = boto.ec2.securitygroup.SecurityGroup(
-        name=go_agent_security_group_name,
-        owner_id=go_agent_security_group_owner,
-        id=go_agent_security_group
+    go_agent_security_group = ec2.create_security_group(
```
Comment (awais786) [inline tubular/admin/boto_to_add_ingress.py:87]: Verify this also.

File: tubular/admin/boto_to_add_ingress.py [hunk 4]
```diff
@@ -45,90 +45,105 @@ def add_ingress_rule(dry_run, go_agent_security_group, go_agent_security_group_o
...
             yes_no = six.moves.input("Apply the change to this security group? [Yes]")
             if yes_no in ("", "y", "Y", "yes"):
-                sec_group.authorize(
-                    ip_protocol='tcp',
-                    from_port=22,
-                    to_port=22,
-                    src_group=go_agent_security_group,
-                    dry_run=dry_run
+                # ssh access
+                ec2.authorize_security_group_ingress(
```
Comment (awais786) [inline tubular/admin/boto_to_add_ingress.py:121]: verify this also.

**Comment by awais786** [comment]
Last PR https://github.com/openedx/tubular/pull/656

**Review by dianakhuang — APPROVED** [review]

**Comment by awais786** [comment]
Since we are catching clienterror at multiple locations so in case of no matching error code raise the exception again.

**Review by ohnickmoy — APPROVED** [review]
Cautionary thumb. On the surface I get the changes, but since I don't know the inner workings of tubular, I'm kinda shaky on giving it the magic thumb 100%

**Pull request merged**